ksqldb-engine/src/main/java/io/confluent/ksql/internal/TelemetryMetricsReporter.java

```diff
@@ -0,0 +1,141 @@
+  // ToDo: How is the cluster ID passed to this class
+  private final String clusterId = "";
+  private final String metricGroup;
+
+
+  public TelemetryMetricsReporter(
+          final String metricGroup,
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config
+  ) {
+    this(metricGroup, providerNamespace, resourceType, config, new TelemetryReporter());
+  }
+
+  protected TelemetryMetricsReporter(
+          final String metricGroup,
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config,
+          final TelemetryReporter reporter
+  ) {
+    this.metricGroup = metricGroup;
+    this.reporter = reporter;
+    configureTelemetryReporter(providerNamespace, resourceType, config, reporter);
+  }
+
+  @Override
+  public void report(final List<DataPoint> dataPoints) {
+    for (final DataPoint dataPoint : dataPoints) {
+      emitMetric(
+              dataPoint.getName(),
+              dataPoint.getTime(),
+              dataPoint.getValue(),
+              ImmutableMap.<String, String>builder()
+                      .putAll(metricTags).putAll(dataPoint.getTags()).build()
+      );
+    }
+  }
+
+  void emitMetric(
+          final String metricName,
+          final Instant timestamp,
+          final Object value,
+          final Map<String, String> metricTags
+  ) {
+    // Todo: Is it correct to use the KsqlProvider provided by ce-kafka?
+    final String fullMetricName = MetricsUtils.fullMetricName(
+            KsqlProvider.DOMAIN, metricGroup, metricName);
+    final MetricKey key = new MetricKey(fullMetricName, metricTags);
+
+    reporter.emitter().emitMetric(SinglePointMetric.gauge(key, (Double) value, timestamp));
+  }
+
+  private void configureTelemetryReporter(
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config,
+          final TelemetryReporter reporter
+  ) {
+    // Todo: Is it correct to use the KsqlProvider provided by ce-kafka?
+    ProviderRegistry.registerProvider(
+            providerNamespace,
+            KsqlProvider.class.getCanonicalName()
+    );
+
+    reporter.configure(config);
+    final MetricsContext ctx = createMetricContext(providerNamespace, resourceType);
+    try {
+      reporter.contextChange(ctx);
+    } catch (final Throwable e) {
+      logger.error("Failed to properly start up telemetry reporter with the following error: ", e);
+      throw e;
+    }
+  }
+
+  private MetricsContext createMetricContext(
+          final String providerNameSpace,
+          final String resourcetype
+  ) {
+    final Map<String, Object> rawConfig = new HashMap<>();
+    rawConfig.put(RESOURCE_LABEL_VERSION, AppInfoParser.getVersion());
+    rawConfig.put(RESOURCE_LABEL_TYPE, resourcetype);
+    rawConfig.put(RESOURCE_LABEL_COMMIT_ID, AppInfoParser.getCommitId());
+    rawConfig.put(RESOURCE_CLUSTER_ID, clusterId);
```

Comment:
Just to double check - right now this will always be a "", right? Also @rodesai talked about wanting node ID but I'm not sure how to get that either, we'd want both cluster ID _and_ node ID, right? Alternatively, node ID can be through the metrics if we have access to it there

Reply:
we should had a node ID tag, but I'm not sure how to have it set correctly - let's sync with the telemetry folks tomorrow

Reply:
I'm also curious --- for node ID, how do we get its value? Would that be related to the pod name or some pre-specified value we would read from elsewhere?